src/NuGet.Protocol.VisualStudio/PSSearchResourceV3.cs

```diff
@@ -1,35 +1,100 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+using NuGet.Packaging.Core;
 using NuGet.Protocol.Core.Types;
+using NuGet.Protocol.Core.v3;
+using NuGet.Versioning;
 
 namespace NuGet.Protocol.VisualStudio
 {
-    public class PSSearchResourceV3 : PSSearchResource
+    public class PowerShellSearchResourceV3 : PSSearchResource
     {
-        private readonly UISearchResourceV3 _searchResource;
+        private readonly RawSearchResourceV3 _searchResource;
 
-        public PSSearchResourceV3(UISearchResourceV3 searchResource)
+        public PowerShellSearchResourceV3(RawSearchResourceV3 searchResource)
         {
             _searchResource = searchResource;
         }
 
-        public override async Task<IEnumerable<PSSearchMetadata>> Search(string search, SearchFilter filters, int skip, int take, CancellationToken token)
+        public override async Task<IEnumerable<PSSearchMetadata>> Search(string searchTerm,
```

Comment:
I wish we can rename to use the 'Async' suffix here and all the other methods in this resource and its base classes. Since this rename will require changes in all the repos, we can do it in a different iteration